addons/csi-openstack-cinder/openstack-csi-cinder.yaml

```diff
@@ -0,0 +1,567 @@
+{{ if .Config.CABundle }}
+{{ caBundleVolumeMount | indent 12 }}
+{{ end }}
+          resources:
+            {}
+      volumes:
+        - name: socket-dir
+          hostPath:
+            path: /var/lib/kubelet/plugins/cinder.csi.openstack.org
+            type: DirectoryOrCreate
+        - name: registration-dir
+          hostPath:
+            path: /var/lib/kubelet/plugins_registry/
+            type: Directory
+        - name: kubelet-dir
+          hostPath:
+            path: /var/lib/kubelet
+            type: Directory
+        # - name: pods-cloud-data
+        #   hostPath:
+        #     path: /var/lib/cloud/data
+        #     type: Directory
+        - name: pods-probe-dir
+          hostPath:
+            path: /dev
+            type: Directory
+        - name: cloud-config
+          secret:
+            secretName: cloud-config
+
+        - hostPath:
+            path: /etc/cacert
+          name: cacert
+{{ if .Config.CABundle }}
+{{ caBundleVolume | indent 8 }}
+{{ end }}
+      affinity:
+        {}
+      nodeSelector:
+        {}
+      tolerations:
+        - operator: Exists
+      priorityClassName: system-cluster-critical
+---
+# Source: openstack-cinder-csi/templates/controllerplugin-deployment.yaml
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: openstack-cinder-csi-controllerplugin
+  namespace: kube-system
+  labels:
+    component: controllerplugin
+    app: openstack-cinder-csi
+    release: cinder-csi
+    chart: openstack-cinder-csi-2.27.1
+    heritage: Helm
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+      maxSurge: 1
+  selector:
+    matchLabels:
+      component: controllerplugin
+      app: openstack-cinder-csi
+      release: cinder-csi
+  template:
+    metadata:
+      labels:
+        component: controllerplugin
+        app: openstack-cinder-csi
+        release: cinder-csi
+        chart: openstack-cinder-csi-2.27.1
+        heritage: Helm
+    spec:
+      serviceAccount: csi-cinder-controller-sa
+      containers:
+        - name: csi-attacher
+          image: {{ .InternalImages.Get "OpenstackCSIAttacher" }}
+          imagePullPolicy: IfNotPresent
+          args:
+            - "-v=1"
+            - "--csi-address=$(ADDRESS)"
+            - "--timeout=3m"
+            - "--leader-election=true"
+            - "--default-fstype=ext4"
+          env:
+            - name: ADDRESS
+              value: /var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+          resources:
+            {}
+        - name: csi-provisioner
+          image: {{ .InternalImages.Get "OpenstackCSIProvisioner" }}
+          imagePullPolicy: IfNotPresent
+          args:
+            - "-v=1"
+            - "--csi-address=$(ADDRESS)"
+            - "--timeout=3m"
+            - "--leader-election=true"
+            - "--default-fstype=ext4"
```

Comment:
This option is only supported with CSI driver v1.26+

Reply:
wrapped back in the condition